microservices/services/contest-service/src/main/java/org/xcolab/service/proposal/domain/proposal/ProposalDaoImpl.java

```diff
@@ -345,4 +363,109 @@ public Integer getProposalMaterializedPoints(Long proposalId) {
 
         return query.fetchInto(Proposal.class);
     }
+
+    public static void deleteOrphanProposals(DSLContext ctx) {
```

Comment:
Why does this need to be static? Putting a static method in a DaoImpl is not nice.

Reply:
It does not use the object's `dslContext` but the provided one and is therefore independent from a single `ProposalDaoImpl` object. That is necessary to allow using the transactions `DSLContext`. The method being `static` reflects that.

Where would you put the method instead?

Reply:
I see. The main issue is that the `ContestDaoImpl` is now closely coupled with the `ProposalDaoImpl` by relying on the specific implementation class. Another consideration is whether there are cases where it makes sense to call `deleteOphanProposals` without providing a specific context. This raises two questions:
* If the `deleteOphanProposals` method must always run inside a transaction, how are we preventing it from being called with a transaction-less context?
* If it makes sense to call `deleteOrphanProposals` with a regular context, shouldn't it be a normal method of the Dao?

It sounds to me like this would be a general problem whenever we need to use transactions.
One potential solution I can think of is to still inject the ProposalDao, but to allow modifying which DSLContext is used for the operations. This could be done with a copy method, e.g. `ProposalDao#withContext(DSLContext dslContext)`:

```java
// class ProposalDaoImpl
@Override
public ProposalDao withContext(DSLContext dslContext) {
    return new ProposalDaoImpl(dslContext);
}
```

The reason for this being a copy method rather than a static factory method is that we want to define it in the `ProposalDao` interface and implement it in `ProposalDaoImpl`. Doing this also allows to add additional injected parameters at a later point, which would be passed to the constructor in the copy method.

How about making it a regular Dao method that wraps itself in a transaction and allowing a custom dslContext through the method above? Not sure how jOOQ handles nested transactions (I don't think MySQL supports them directly), but we could try this out.